packages/sdk/src/oracle/services/statemachines/fetchPastEvents.ts

```diff
@@ -54,11 +54,11 @@ export function Handlers(store: Store): GenericHandlers<Params, Memory> {
         // the provider threw an error so we will reduce our range by moving startblock closer to endblock next iteration
         memory.state = rangeFailureDescending(rangeState);
       }
-      // the range functions will tell us when we have successfully queried the entire range of blocks.
-      if (done) return "done";
       memory.iterations++;
+      // the range functions will tell us when we have successfully queried the entire range of blocks.
+      if (memory?.state?.done) return "done";
       // sleep to let other contexts run, but just resume right after.
-      return ctx.sleep(10);
+      return ctx.sleep(100);
```

Comment:
magic numbers, no real reason other than it should iterate pretty quick, but slowed it down to 100 ms just in case theres some kind of issue causing it to keep looping, 10x less spam. but so far have not seen a problem

Reply:
100ms seems fast? Do we need updates at that rate?